File: src/main/java/phoenix/enity/tasks/MoveRandomGoal.java
package phoenix.enity.tasks;

import net.minecraft.entity.ai.goal.Goal;
import net.minecraft.entity.monster.VexEntity;
import net.minecraft.util.math.BlockPos;
import phoenix.enity.TalpaEntity;

import java.util.EnumSet;

public class MoveRandomGoal extends Goal
{
    TalpaEntity entity;
    public MoveRandomGoal(TalpaEntity entity) {
        this.setMutexFlags(EnumSet.of(Goal.Flag.MOVE));
        this.entity = entity;
    }

    /**
     * Returns whether execution should begin. You can also read and cache any state necessary for execution in this
     * method as well.
     */
    public boolean shouldExecute() {
        return !entity.getMoveHelper().isUpdating() && entity.getRNG().nextInt(7) == 0;
    }

    /**
     * Returns whether an in-progress EntityAIBase should continue executing
     */
    public boolean shouldContinueExecuting() {
        return false;
    }

    /**
     * Keep ticking a continuous task that has already been started
     */
    public void tick()
    {
        BlockPos blockpos = entity.getBoundOrigin();
        if (blockpos == null)
        {
            blockpos = new BlockPos(entity);
        }

        for(int i = 0; i < 3; ++i)
        {
            BlockPos blockpos1 = blockpos.add(entity.getRNG().nextInt(15) - 7, entity.getRNG().nextInt(11) - 5, entity.getRNG().nextInt(15) - 7);
            if (entity.world.isAirBlock(blockpos1))
            {
                entity.getMoveHelper().setMoveTo((double)blockpos1.getX() + 0.5D, (double)blockpos1.getY() + 0.5D, (double)blockpos1.getZ() + 0.5D, 0.25D);
                if (entity.getAttackTarget() == null)
                {
                    entity.getLookController().setLookPosition((double)blockpos1.getX() + 0.5D, (double)blockpos1.getY() + 0.5D, (double)blockpos1.getZ() + 0.5D, 180.0F, 20.0F);
                }
                break;
            }
        }

    }
}

File: src/main/java/phoenix/enity/tasks/MoveRandomGoal.kt
package phoenix.enity.tasks

import net.minecraft.entity.ai.goal.Goal
import net.minecraft.util.math.BlockPos
import phoenix.enity.TalpaEntity
import java.util.*


class MoveRandomGoal(entity: TalpaEntity) : Goal()
{
    var entity: TalpaEntity

    /**
     * Returns whether execution should begin. You can also read and cache any STATE necessary for execution in this
     * method as well.
     */
    override fun shouldExecute(): Boolean = !entity.moveHelper.isUpdating && entity.rng.nextInt(7) == 0

    /**
     * Returns whether an in-progress EntityAIBase should continue executing
     */
    override fun shouldContinueExecuting(): Boolean = false

    /**
     * Keep ticking a continuous task that has already been started
     */
    override fun tick()
    {
        var blockpos = entity.boundOrigin
        if (blockpos == null)
        {
            blockpos = BlockPos(entity)
        }
        for (i in 0..2)
        {
            val randompoint =
                blockpos.add(entity.rng.nextInt(15) - 7, entity.rng.nextInt(11) - 5, entity.rng.nextInt(15) - 7)
            if (entity.world.isAirBlock(randompoint))
            {
                entity.moveHelper.setMoveTo(
                    randompoint.x.toDouble() + 0.5, randompoint.y.toDouble() + 0.5, randompoint.z
                        .toDouble() + 0.5, 0.25
                )
                if (entity.attackTarget == null)
                {
                    entity.lookController.setLookPosition(
                        randompoint.x.toDouble() + 0.5, randompoint.y.toDouble() + 0.5, randompoint.z
                            .toDouble() + 0.5, 180.0f, 20.0f
                    )
                }
                break
            }
        }
    }

    init
    {
        this.mutexFlags = EnumSet.of(Flag.MOVE)
        this.entity = entity
    }
}
